core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -122,7 +126,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
         } else {
           logInfo("Registered executor: " + executorRef + " with ID " + executorId)
           context.reply(RegisteredExecutor)
-
+          latestTokens.foreach(x => context.reply(NewTokens(x)))
```

Comment:
My concern with `send` (or with your current `reply`) is that the sender doesn't know if it arrived. So you could have `RegisteredExecutor` arrive but `NewTokens` not arrive. My hope is that in that case the connection would be terminated and the executor would go away (instead of staying up in a broken state), but with akka, I have no idea.